include/deal.II/base/config.h.in

```diff
@@ -357,6 +357,8 @@
     DEAL_II_MPI_VERSION_MINOR) \
     >=  \
     (major)*100 + (minor))
+#else
+#  define DEAL_II_MPI_VERSION_GTE(major,minor) false
```

Comment:
Separately: The preprocessor does not know the symbol `false`. You will end up with 
```
  #if false
```
which the compiler interprets in the following way: Since the preprocessor doesn't know about the token `false`, it replaces it with `0`, so you get
```
  #if 0
```
This has the same effect of course as saying `if false`, but more by accident: if you had done
```
  #else
  #  define DEAL_II_MPI_VERSION_GTE(major,minor) true
```
then `true` would also be interpreted as `0`, i.e. "false".

Reply:
> I don't think that we should give a symbol a value when no value can reasonably be defined.

In my opinion it is nevertheless useful. In your example you distinguish between without MPI and with MPI (with two versions: MPI-2 and with MPI-3). I would however like to be able to group these options differently with MPI-3 and without MPI-3 (without MPI or with MPI-2). This grouping enables me significant simpler code.

Regarding the second point `#if false`/`#if true`: my experiments shows that the compiler understands these correctly. Do you have an example where this is not the case?